### PR TITLE
Release 6

### PR DESCRIPTION
**Pull request by Mikmist**
Hereby the sixth sprint, with changes to most fundamental parts of the app under which auth0 support. New UI, less wasted space and more. 

Note: The PR is a bit late because my internet was absolute garbage in the train, my apologies.

### CODE REVIEW

File: Hestia/Hestia/backend/utils/PingServer.cs [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-        }
-    }
+﻿using Hestia.backend.exceptions;
+using System;
+
+namespace Hestia.backend.utils
+{
+    class PingServer
+    {
+        public static bool Check(string address, int port)
```
Comment (Feikoritsema) [inline Hestia/Hestia/backend/utils/PingServer.cs]: Definitely better :)

File: Hestia/Hestia/backend/utils/server_discovery/AutoServerDicovery.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿using Foundation;
+using System;
+using UIKit;
+
+namespace Hestia.backend.utils.server_discovery
+{
+    /// <summary>
```
Comment (Feikoritsema) [inline Hestia/Hestia/backend/utils/server_discovery/AutoServerDicovery.cs]: More of this documentation in you code would be nice, you have some big classes which could benefit from some better documentation.

File: Hestia/Hestia/frontend/DevicesScreen/TableSourceDevicesMain.cs [hunk 3]
```diff
@@ -96,37 +102,41 @@ public override UITableViewCell GetCell(UITableView tableView, NSIndexPath index
                 //    {
                 //        cell.ImageView.Image = UIImage.FromBundle("Images/lock.png");
```
Comment (Feikoritsema) [inline Hestia/Hestia/frontend/DevicesScreen/TableSourceDevicesMain.cs]: commented out

File: Hestia/Hestia/frontend/EntryScreen/UIViewControllerAuth0.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+using Hestia.Resources;
+using Auth0.OidcClient;
+using Hestia.backend.authentication;
+using System.Threading.Tasks;
+using IdentityModel.OidcClient;
+using System.Diagnostics;
+
+namespace Hestia
+{
+	// TODO remove if local/global is finished
```
Comment (Feikoritsema) [inline Hestia/Hestia/frontend/EntryScreen/UIViewControllerAuth0.cs]: Especially for the last sprint submission: no TODO and commented out code. please make sure

File: Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                    Console.WriteLine("To serverconnect");
+                    PerformSegue(strings.mainToServerConnect, this);
+                }
+            };
+
+            ToGlobalButton.TouchUpInside += async delegate (object sender, EventArgs e)
+            {
+                if (HasValidGlobalLogin())
+                {
+					SetValuesAndSegueToServerSelectGlobal();
```
Comment (Feikoritsema) [inline Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs]: weird indentation

File: Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            if (defaultAccessToken != null)
+            {
+                NetworkHandler networkHandler = new NetworkHandler(Resources.strings.webserverIP, defaultAccessToken);
+                try
+                {
+                    HestiaWebServerInteractor hestiaWebServerInteractor = new HestiaWebServerInteractor(networkHandler);
+                }
+                catch (Exception ex)
+                {
+                    Console.Write(ex.StackTrace);
```
Comment (Feikoritsema) [inline Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs]: Be consistent in printing stack trace or toString

File: Hestia/Hestia/frontend/Firebase/ServerSelectScreen/TableSourceServerList.cs [hunk 7]
```diff
@@ -28,23 +28,23 @@ public override nint NumberOfSections(UITableView tableView)
 
 		public override nint RowsInSection(UITableView tableview, nint section)
 		{
-            return Globals.FirebaseServers.Count;
+            return Globals.Auth0Servers.Count;
```
Comment (Feikoritsema) [inline Hestia/Hestia/frontend/Firebase/ServerSelectScreen/TableSourceServerList.cs]: Indentation here looks awful, but I think that's just on github.

File: Hestia/Hestia/AppDelegate.cs [hunk 8]
```diff
@@ -1,59 +1,219 @@
...
+        {
+            Xamarin.Calabash.Start();
+
+            Globals.ScreenHeight = (int)UIScreen.MainScreen.Bounds.Height;
+            Globals.ScreenWidth = (int)UIScreen.MainScreen.Bounds.Width;
+            // Override point for customization after application launch.
+            // If not required for your application you can safely delete this method
+            NSUserDefaults userDefaults = NSUserDefaults.StandardUserDefaults;
+
+            // For Debugging
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/AppDelegate.cs]: The appdelegate was indeed still a bit messy. In 0f814abbe7a4ee06c5286e2b8c985b54bcfdf194 it is cleaned up.

File: Hestia/Hestia/AppDelegate.cs [hunk 9]
```diff
@@ -1,59 +1,219 @@
...
+                    Window.RootViewController = uIViewControllerLocalGlobal;
+                    uIViewControllerLocalGlobal.CalledFromAppDelegateAsync();
+                }
+                Window.MakeKeyAndVisible();
+            }
+
+            return true;
+        }
+
+	public override void OnResignActivation(UIApplication application)
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/AppDelegate.cs]: I think they were in the autogenerated project file. We now only override the FinishedLaunching, so indeed the others are removed in 0f814abbe7a4ee06c5286e2b8c985b54bcfdf194

File: Hestia/Hestia/AppDelegate.cs [hunk 10]
```diff
@@ -1,59 +1,219 @@
...
+            set;
+        }
+
+        public static UIStoryboard mainStoryboard = UIStoryboard.FromName(strings.mainStoryBoard, null);
+        public static UIStoryboard devices2Storyboard = UIStoryboard.FromName(strings.devices2StoryBoard, null);
+
+        string defaultServername;
+        string defaultIP;
+        string defaultPort;
+        string defaultAuth0AccessToken;
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/AppDelegate.cs]: This is also fixed in 0f814abbe7a4ee06c5286e2b8c985b54bcfdf194

File: Hestia/Hestia/AppDelegate.cs [hunk 11]
```diff
@@ -1,59 +1,219 @@
...
+                else
+                {
+                    Window.RootViewController = serverConnectViewController;
+                }
+                Window.MakeKeyAndVisible();
+            }
+            else
+            {
+                Globals.LocalLogin = false;
+                //UIViewControllerAuth0 auth0ViewController = mainStoryboard.InstantiateViewController(strings.auth0ViewController) as UIViewControllerAuth0;
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/AppDelegate.cs]: Removed in 0f814abbe7a4ee06c5286e2b8c985b54bcfdf194

File: Hestia/Hestia/frontend/DevicesScreen/AddDeviceScreen/PropertyCell.cs [hunk 12]
```diff
@@ -18,7 +18,7 @@ public PropertyCell(NSString cellId) : base(UITableViewCellStyle.Default, cellId
         // Sets the placeholder from the required info
         public void UpdateCell(string placeholder)
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/frontend/DevicesScreen/AddDeviceScreen/PropertyCell.cs]: This was indeed not used anymore in the new design of the properties screen. See c323e33ad4056e10415943d4fa151976fcfd3ae9

File: Hestia/Hestia/frontend/Local/UITableViewControllerLocalLogin.cs [hunk 13]
```diff
@@ -94,4 +93,4 @@ private bool TextFieldShouldReturn(UITextField textfield)
             return false; //No line-breaks.
         }
     }
```
[the remark this answers is not in the document]
Comment (MarcF12) [reply Hestia/Hestia/frontend/Local/UITableViewControllerLocalLogin.cs]: This class should have been removed now the local login is dropped. It is gone in 67b58d01ca3fb97861294bceb7136535a42cfbe1

File: Hestia/Hestia/frontend/DevicesScreen/TableSourceDevicesMain.cs (hunk 3 above)
Comment (MarcF12) [reply Hestia/Hestia/frontend/DevicesScreen/TableSourceDevicesMain.cs]: Not sure if we are still going to use it, for now removed in 66c8a1fb4f8e4adba71d6222e28fe493127537af

File: Hestia/Hestia/frontend/EntryScreen/UIViewControllerAuth0.cs (hunk 4 above)
Comment (MarcF12) [reply Hestia/Hestia/frontend/EntryScreen/UIViewControllerAuth0.cs]: The functionality of this ViewController was replaced by UIViewControllerLocalGlobal. So it is gone in a8201d215aed3690f28a8903a1af3333351ae246

File: Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs (hunk 5 above)
Comment (MarcF12) [reply Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs]: Hmm was not visible in Visual Studio, but I think it is fixed in a8201d215aed3690f28a8903a1af3333351ae246

File: Hestia/Hestia/frontend/Firebase/ServerSelectScreen/TableSourceServerList.cs (hunk 7 above)
Comment (MarcF12) [reply Hestia/Hestia/frontend/Firebase/ServerSelectScreen/TableSourceServerList.cs]: Was also not visible in Visual Studio, but should be fixed in e5e31a2aabbf4b7159da381fbe22dbdba1baf630

**Comment by MarcF12** [comment]
Thanks for the feedback. I think most of it is adressed now. @daangroot Can you have a look at the remaining back end comments? Regarding "When I click on local server there's no way to go back to the main screen.", I think there was a possibility to go back via the settings screen, but it was indeed not user friendly. So I added a cancel button in one of the commits.

**Comment by Feikoritsema** [comment]
Thanks for all the changes. For the final sprint (and the wrap-up sprint) make sure you code is as clean as it can be, and all the documentation is fully up to date and neat.

**Pull request merged**